### PR TITLE
Switch to scope activationHook from language package

### PR DESCRIPTION
**Pull request by vzamanillo**
https://github.com/AtomLinter/Meta/issues/20

### CODE REVIEW

File: lib/linter-reek.js [hunk 1]
```diff
@@ -2,20 +2,10 @@
 
 // eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
 import { CompositeDisposable } from 'atom';
+import * as helpers from 'atom-linter';
```
Comment (Arcanemagus) [inline lib/linter-reek.js]: When the package gets _activated_ is controlled by the `activationHook`, however the package is still _loaded_ on Atom startup. These should remain lazy loaded as before, however the deferred call in `activate` is no longer meaningful.

File: package.json [hunk 2]
```diff
@@ -13,6 +13,10 @@
       "default": "reek"
     }
   },
+  "activationHooks": [
+    "language-ruby:grammar-used",
```
Comment (Arcanemagus) [inline package.json]: The point of https://github.com/AtomLinter/Meta/issues/20 is to use the `scope.name:root-scope-used` hook, _specifically_ to avoid the need to list out the name of all the different grammar packages that provide that scope.

File: lib/linter-reek.js [hunk 3]
```diff
@@ -52,21 +52,12 @@ const parseJSON = (stdout) => {
...
-      loadDeps();
-      if (atom.inDevMode()) {
-        // eslint-disable-next-line no-console
-        console.log('linter-reek: All dependencies installed.');
-      }
-    };
-    depsCallbackID = window.requestIdleCallback(installLinterReekDeps);
-    this.idleCallbacks.add(depsCallbackID);
+    require('atom-package-deps').install('linter-reek', true);
+    loadDeps();
```
Comment (Arcanemagus) [inline lib/linter-reek.js]: Just did some local testing on this, and it looks like the file itself (`linter-reek.js`) is only loaded when the activation hook is triggered... but it **blocks** the opening of the file until the package is done calling `activate()`.

This means that all the optimizations we were doing before to reduce the load during Atom's initial startup are still useful for packages that use `activationHooks` because we don't want to be blocking the UI any more than we must.

To put it another way:
Move all of this back into the `idleCallback`. 😛
